test/module/irohad/ametsuchi/block_query_transfer_test.cpp

```diff
@@ -77,20 +82,25 @@ namespace iroha {
      * @then query returns the transaction
      */
     TEST_F(BlockQueryTransferTest, SenderAssetName) {
-      model::Block block;
-      block.transactions.emplace_back();
-
-      block.transactions.back().commands.push_back(
-          cmd_gen.generateTransferAsset(creator1, creator2, asset, {}));
-      tx_hashes.push_back(iroha::hash(block.transactions.back()));
-
-      block.height = 1;
+      auto block =
+          TestBlockBuilder()
+              .transactions(std::vector<shared_model::proto::Transaction>(
+                  {TestTransactionBuilder()
+                       .transferAsset(
+                           creator1, creator2, asset, "Transfer asset", "0.0")
+                       .build()}))
+              .height(1)
+              .prevHash(shared_model::crypto::Hash(std::string("0", 32)))
+              .txNumber(1)
+              .build();
```

Comment:
Extract block building routine as a function

Reply:
What is the reason? You need to pass all of the parameters anyway

Reply:
Code duplication, these blocks are same across the tests